### PR TITLE
[Clang-tidy header][24/N] Fix clang-tidy warnings on c10/cuda/*.{cpp,h}

### PR DESCRIPTION
**Pull request by cyyever**
This PR begins to clean clang-tidy warnings of code in c10/cuda.

### CODE REVIEW

File: c10/cuda/CUDAAllocatorConfig.cpp [hunk 1]
```diff
@@ -118,9 +119,9 @@ size_t CUDAAllocatorConfig::parseRoundUpPower2Divisions(
   bool first_value = true;
 
   if (++i < config.size()) {
-    if (config[i].compare("[") == 0) {
+    if (std::string_view(config[i]) == "[") {
       size_t last_index = 0;
-      while (++i < config.size() && config[i].compare("]") != 0) {
+      while (++i < config.size() && std::string_view(config[i]) != "]") {
```
Comment (Skylion007) [inline c10/cuda/CUDAAllocatorConfig.cpp:124]: Can you just compare the chars directly here? Wouldn't that be more efficient?

File: c10/cuda/CUDAAllocatorConfig.h [hunk 2]
```diff
@@ -108,7 +108,7 @@ class C10_CUDA_API CUDAAllocatorConfig {
       size_t i);
 
   std::atomic<size_t> m_max_split_size;
-  std::vector<size_t> m_roundup_power2_divisions;
+  std::vector<size_t> m_roundup_power2_divisions{};
```
Comment (Skylion007) [inline c10/cuda/CUDAAllocatorConfig.h:111]: I'm confused why this is necessary.

Comment (cyyever) [reply c10/cuda/CUDAAllocatorConfig.h]: Not necessary, reverted.

**Comment by cyyever** [comment]
@Skylion007 Ping you for reviewing.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**